### PR TITLE
nixosTests.prosody*: port to python

### PR DESCRIPTION
**Pull request by globin**
###### Motivation for this change
#72828

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by globin** [comment]
@GrahamcOfBorg test prosody prosodyMysql

File: nixos/tests/xmpp/prosody.nix [hunk 1]
```diff
@@ -28,19 +28,19 @@ import ../make-test.nix {
   };
 
   testScript = { nodes, ... }: ''
-    $server->waitForUnit('prosody.service');
-    $server->succeed('prosodyctl status') =~ /Prosody is running/;
+    server.wait_for_unit("prosody.service")
+    server.succeed('prosodyctl status | grep "Prosody is running"')
```
Comment (worldofpeace) [inline nixos/tests/xmpp/prosody.nix]: could we do something like `'Prosody is running' in server.succeed('prosodyctl status')`

Comment (globin) [reply nixos/tests/xmpp/prosody.nix]: That would not be enough to make the test fail, I think this is simpler

**Review by worldofpeace — APPROVED** [review]
Tests succeed for me.

**Pull request merged**